python/lsst/cp/pipe/ptc.py

```diff
@@ -706,122 +420,338 @@ def measureMeanVarCov(self, exposure1, exposure2, region=None, covAstierRealSpac
+        max=1.0,
+    )
+    minMeanRatioTest = pexConfig.Field(
+        dtype=float,
+        doc="In the initial test to screen out bad points with a ratio test, points with low"
+            " flux can get inadvertantly screened.  This test only screens out points with flux"
+            " above this value.",
+        default=20000,
+    )
+    minVarPivotSearch = pexConfig.Field(
+        dtype=float,
+        doc="The code looks for a pivot signal point after which the variance starts decreasing at high-flux"
+            " to exclude then form the PTC model fit. However, sometimes at low fluxes, the variance"
+            " decreases slightly. Set this variable for the variance value, in ADU^2, after which the pivot "
+            " should be sought.",
+        default=10000,
+    )
+    doFitBootstrap = pexConfig.Field(
+        dtype=bool,
+        doc="Use bootstrap for the PTC fit parameters and errors?.",
+        default=False,
+    )
+
 
-        return mu, varDiff, covDiffAstier
+class PhotonTransferCurveSolveTask(pipeBase.PipelineTask,
+                                   pipeBase.CmdLineTask):
+    """Task to fit the PTC from flat covariances.
+    """
+    ConfigClass = PhotonTransferCurveSolveConfig
+    _DefaultName = 'cpPhotonTransferCurveSolve'
 
-    def computeCovDirect(self, diffImage, weightImage, maxRange):
-        """Compute covariances of diffImage in real space.
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        """Ensure that the input and output dimensions are passed along.
+        Parameters
+        ----------
+        butlerQC : `lsst.daf.butler.butlerQuantumContext.ButlerQuantumContext`
+            Butler to operate on.
+        inputRefs : `lsst.pipe.base.connections.InputQuantizedConnection`
+            Input data refs to load.
+        ouptutRefs : `lsst.pipe.base.connections.OutputQuantizedConnection`
+            Output data refs to persist.
+        """
+        inputs = butlerQC.get(inputRefs)
+        outputs = self.run(inputCovariances=inputs['inputCovariances'], camera=inputs['camera'])
+        butlerQC.put(outputs, outputRefs)
 
-        For lags larger than ~25, it is slower than the FFT way.
-        Taken from https://github.com/PierreAstier/bfptc/
+    def run(self, inputCovariances, camera=None, inputDims=None, outputDims=None):
+        """Fit measure covariances to different models.
 
         Parameters
         ----------
-        diffImage : `numpy.array`
-            Image to compute the covariance of.
+        inputCovariances : `list` [`lsst.ip.isr.PhotonTransferCurveDataset`]
+            List of lsst.ip.isr.PhotonTransferCurveDataset datasets.
+
+        camera : `lsst.afw.cameraGeom.Camera`, optional
+            Input camera.
 
-        weightImage : `numpy.array`
-            Weight image of diffImage (1's and 0's for good and bad pixels, respectively).
+        inputDims : `list` [`lsst.daf.butler.DataCoordinate`], optional
+            DataIds to use.
```

Comment:
Im not sure I would know what I was reading with the name inputDims, is there a more descriptive variable name?

Reply:
I changed the name to `inputExpList`.